kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java

```diff
@@ -724,5 +762,10 @@ private static String connectionTypeConvert(NMDeviceType deviceType) {
                     .format("Unsupported connection type conversion from NMDeviceType \"%s\"", deviceType.toString()));
         }
     }
+    
+    private static Integer getIp4MtuFromProperties(NetworkProperties properties, String deviceId) {
+        Optional<Integer> mtu = properties.getOpt(Integer.class, "net.interface.%s.config.ip4.mtu", deviceId);
+        return mtu.orElse(0);
+    }
```

Comment:
This was to let a user delete the mtu from its field and let it reset to 0/auto on its own, but your suggestion is also fine :)
`getIp4MtuFromProperties` will return an Optional (I'd keep the method as its called from 3 different places) and as such will follow the usual map population logic

Reply:
> I think you need this is caused by [this code here](https://github.com/eclipse/kura/blob/fc539f270f41906058e2fff72401479e75bd48ee/kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java#L110-118) where you'll have empty map if no mtu is set.
> 
> I would check there if the map is empty and avoid setting the map entirely in the higher level 🤔 What do you think?

Can confirm that Network Manager doesn't care about empty maps, so there's no need for additional checks 😋 
Should the UI bring up a 0 as a default value or just leave an empty field until otherwise configured?

Reply:
> Should the UI bring up a 0 as a default value or just leave an empty field until otherwise configured?

0 as default. We had a field set as your second option in the past, but decided it was confusing for the user (it was the WAN priority, which behaves similarly to MTU)